RemoveUnreachableBlocksStep causes NotSupportedException during dotnet/runtime build

See https://dev.azure.com/dnceng/public/_build/results?buildId=464367&view=logs&jobId=1136f1e1-214d-5201-e246-afde0074522e&j=1136f1e1-214d-5201-e246-afde0074522e&t=22aca3ba-752a-5db5-8c5e-3f034e37316c from https://github.com/dotnet/runtime/pull/1088

```
  Fatal error in IL Linker
  Unhandled exception. System.NotSupportedException: Missing predefined 'System.Object' type
     at Mono.Linker.Steps.RemoveUnreachableBlocksStep.BodyReducer.CleanRemovedVariables(List`1 variables)
     at Mono.Linker.Steps.RemoveUnreachableBlocksStep.BodyReducer.RewriteBody()
     at Mono.Linker.Steps.RemoveUnreachableBlocksStep.RewriteBody(MethodDefinition method)
     at Mono.Linker.Steps.RemoveUnreachableBlocksStep.RewriteBodies(Collection`1 types)
     at Mono.Linker.Steps.RemoveUnreachableBlocksStep.Process()
     at Mono.Linker.Steps.BaseStep.Process(LinkContext context)
     at Mono.Linker.Pipeline.ProcessStep(LinkContext context, IStep step)
     at Mono.Linker.Pipeline.Process(LinkContext context)
     at Mono.Linker.Driver.Run(ILogger customLogger)
     at Mono.Linker.Driver.Execute(String[] args, ILogger customLogger)
     at Mono.Linker.Driver.Main(String[] args)
/__w/3/s/eng/illink.targets(132,5): error MSB6006: "dotnet" exited with code 134. [/__w/3/s/src/libraries/System.Text.Encoding.CodePages/src/System.Text.Encoding.CodePages.csproj]
##[error]eng/illink.targets(132,5): error MSB6006: (NETCORE_ENGINEERING_TELEMETRY=Build) "dotnet" exited with code 134.
```

/cc @marek-safar

--- a/src/linker/Linker/BCL.cs
+++ b/src/linker/Linker/BCL.cs
@@ -60,21 +60,26 @@ namespace Mono.Linker
 			return true;
 		}
 
+		static readonly string[] corlibNames = new [] {
+			"mscorlib",
+			"System.Runtime",
+			"System.Private.CoreLib",
+			"netstandard"
+		};
+
 		public static TypeDefinition FindPredefinedType (string ns, string name, LinkContext context)
 		{
 			var cache = context.Resolver.AssemblyCache;
 
-			AssemblyDefinition corlib;
-			TypeDefinition type = null;
-			if (cache.TryGetValue ("mscorlib", out corlib)) {
-				type = corlib.MainModule.GetType (ns, name);
+			foreach (var corlibName in corlibNames) {
+				if (!cache.TryGetValue (corlibName, out AssemblyDefinition corlib))
+					continue;
+
+				TypeDefinition type = corlib.MainModule.GetType (ns, name);
 				// The assembly could be a facade with type forwarders, in which case we don't find the type in this assembly.
 				if (type != null)
 					return type;
 			}
-
-			if (cache.TryGetValue ("System.Private.CoreLib", out corlib))
-				return corlib.MainModule.GetType (ns, name);
 
 			return null;
 		}

